src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/BitcoinHeaderVersionRule.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Base.Deployments;
+using Stratis.Bitcoin.Consensus.Rules;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.Consensus.Rules.CommonRules
+{
+    /// <summary>
+    /// Checks if <see cref="BitcoinMain"/> network block's header has a valid block version.
+    /// </summary>
+    [HeaderValidationRule(CanSkipValidation = true)]
+    public class BitcoinHeaderVersionRule : HeaderVersionRule
+    {
+        /// <inheritdoc />
+        /// <exception cref="ConsensusErrors.BadVersion">Thrown if block's version is outdated or otherwise invalid.</exception>
+        public override Task RunAsync(RuleContext context)
+        {
+            Guard.NotNull(context.ConsensusTip, nameof(context.ConsensusTip));
+
+            ChainedHeader chainedHeader = context.ValidationContext.ChainedHeader;
```

Comment:
Should you not be calling the method ComputeBlockVersion here as well?

Reply:
I'm not sure that would give valid results. At any given stage in the chain's existence, there will be blocks getting mined by miners that follow different sets of BIPs and promote or oppose different BIP9 soft fork proposals. Therefore ComputeBlockVersion can only really be used for block creation, validation has to be a lot less strict.

Reply:
At least we should mimic what bitcoin core does:

```
    // Reject outdated version blocks when 95% (75% on testnet) of the network has upgraded:
    // check for version 2, 3 and 4 upgrades
    if((block.nVersion < 2 && nHeight >= consensusParams.BIP34Height) ||
       (block.nVersion < 3 && nHeight >= consensusParams.BIP66Height) ||
       (block.nVersion < 4 && nHeight >= consensusParams.BIP65Height))
```

Reply:
@aprogiena This was already being done in `BitcoinActivationRule`, should it be moved?

Reply:
Oh, in that case it looks correct. If it is separated, someone did that for a reason I guess. Maybe I would just add `<seealso>` to the description of this rule to that rule so that people are not confused just like I was that something is missing here